File: recognition/src/main/java/br/com/alexandre/friday/recognize/chain/NeighborhoodChain.java
package br.com.alexandre.friday.recognize.chain;

import java.util.Arrays;
import java.util.Collection;

import br.com.alexandre.friday.recognize.RecognizerChain;
import br.com.alexandre.friday.recognize.RecognizerRequest;

public class NeighborhoodChain extends RecognizerChain {
    public Collection<String> execute(RecognizerRequest request) {
        if (request.getLocality() == null || request.getLocality().trim().isEmpty()) {
            return Arrays.asList(String.format("No momento não consegui verificar o bairro, %s!", request.getName()));
        } else {
            return Arrays.asList(String.format("Você está no bairro %s", request.getSubLocality()));
        }
    }

    public boolean apply(RecognizerRequest request) {
        return  (request.getQ().contains("qual") &&
                 request.getQ().contains("é") &&
                 request.getQ().contains("esse") &&
                 request.getQ().contains("bairro"))
                ||
                (request.getQ().contains("que") &&
                 request.getQ().contains("bairro") &&
                 request.getQ().contains("é") &&
                 request.getQ().contains("esse"));
    }
}

File: recognition/src/main/java/br/com/alexandre/friday/recognize/chain/NeighborhoodChain.kt
package br.com.alexandre.friday.recognize.chain

import br.com.alexandre.friday.recognize.RecognizerChain
import br.com.alexandre.friday.recognize.RecognizerRequest
import java.util.*

class NeighborhoodChain: RecognizerChain() {
    override fun execute(request: RecognizerRequest?): MutableCollection<String> {
        return Arrays.asList(String.format("Você está no bairro %s", request?.subLocality))
    }

    override fun apply(request: RecognizerRequest?): Boolean {
        return request?.q!!.contains("qual") &&
                request.q!!.contains("é") &&
                request.q!!.contains("esse") &&
                request.q!!.contains("bairro")
                ||
                request.q!!.contains("que") &&
                request.q!!.contains("bairro") &&
                request.q!!.contains("é") &&
                request.q!!.contains("esse")
    }
}